src/v/cloud_storage/remote_segment_index.cc

```diff
@@ -36,16 +36,21 @@ offset_index::offset_index(
   , _min_file_pos_step(file_pos_step) {}
 
 void offset_index::add(
-  model::offset rp_offset, kafka::offset kaf_offset, int64_t file_offset) {
+  model::offset rp_offset,
+  kafka::offset kaf_offset,
+  int64_t file_offset,
+  model::timestamp max_timestamp) {
     auto ix = index_mask & _pos++;
     _rp_offsets.at(ix) = rp_offset();
     _kaf_offsets.at(ix) = kaf_offset();
     _file_offsets.at(ix) = file_offset;
+    _time_offsets.at(ix) = max_timestamp.value();
     try {
         if ((_pos & index_mask) == 0) {
             _rp_index.add(_rp_offsets);
             _kaf_index.add(_kaf_offsets);
             _file_index.add(_file_offsets);
+            _time_index.add(_time_offsets);
```

Comment:
doesn't _time_index need to be added in the catch(...) below?

Reply:
Good catch, fixed.